src/main/cpp/audioengine.cpp

```diff
@@ -441,11 +473,11 @@ namespace MWEngine {
 
                 // and perform a fail-safe check in case we're exceeding the headroom ceiling
 
-                if ( sample < -1.0 )
-                    sample = -1.0F;
+                if ( sample < -MAX_OUTPUT )
+                    sample = -MAX_OUTPUT;
 
-                else if ( sample > +1.0 )
-                    sample = +1.0F;
+                else if ( sample > +MAX_OUTPUT )
+                    sample = +MAX_OUTPUT;
 
```

Comment:
Noted on certain device that this would lead to audible clipping, keep the headroom slightly below the maximum.